pkg/controller/notification/notification_context.go

```diff
@@ -40,7 +42,11 @@ func NewNotificationContext(client client.Client, userID, namespace string) (*No
 	}
 
 	if emailLbl, exists := instance.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]; exists {
-		notificationCtx.Email = emailLbl
+		notificationCtx.UserEmail = emailLbl
+	}
+
+	if config != nil {
+		notificationCtx.RegistrationURL = config.GetRegistrationServiceURL()
```

Comment:
shouldn't you return an error if `config==nil`?

Reply:
good idea! changed as part of: 057eb87